src/unstructured_client/utils/_decorators.py

```diff
@@ -39,8 +45,29 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> None:
             if url_is_in_kwargs:
                 kwargs["server_url"] = urlunparse(cleaned_url)
             else:
-                args = args[:SERVER_URL_ARG_IDX] + (urlunparse(cleaned_url),) + args[SERVER_URL_ARG_IDX+1:] # type: ignore
-        
+                args = args[:SERVER_URL_ARG_IDX] + (urlunparse(cleaned_url),) + args[SERVER_URL_ARG_IDX + 1 :]  # type: ignore
+
         return func(*args, **kwargs)
 
     return wrapper
+
+
+def suggest_defining_url_if_401(
+    func: Callable[_P, operations.PartitionResponse]
+) -> Callable[_P, operations.PartitionResponse]:
```

Comment:
Every module, class, method, and function should have a docstring. Use it to articulate for yourself what the contract of that method or function is or what the role of a module or class is. All docstrings should be PEP 257 compliant, like:
```python
def func(...):
    """Concise single-line statement of contract.
{blank line}
    More detail when necessary to do things like describe when a function with an
    Optional[..] return value returns None instead of that value, etc.
    """
```

This particular decorator has plenty worth saying about it, like that it's used to patch generated code, needs to be applied by hand every time, what exactly if fixes about the generated code, etc. Document your work for others who follow.